### PR TITLE
Issue (#38): a11y improvements

### PR DESCRIPTION
**Pull request by niiyeboah**
- Add `aria-controls` attr to toggle button
- Change content part to `section` element
- Add `_getContentId` method to create unique content ids
- Add `id` attr to content part

Closes (#38)

### CODE REVIEW

File: src/vaadin-details.js [hunk 1]
```diff
@@ -136,6 +137,7 @@ class DetailsElement extends ControlStateMixin(ElementMixin(ThemableMixin(Polyme
 
   ready() {
     super.ready();
+    this._contentId = this._getContentId();
```
Comment (web-padawan) [inline src/vaadin-details.js:140]: Please align this with the implementation of [`vaadin-radio-group`](https://github.com/vaadin/vaadin-radio-button/blob/ac80d5e6cac1b266c44da81bac79a1c13e930d15/src/vaadin-radio-group.js#L263-L264):

```js
    const uniqueId = (DetailsElement._uniqueId = 1 + DetailsElement._uniqueId || 0);
    this. _contentId = `${this.constructor.is}-content-${uniqueId}`;
```

Note the `DetailsElement` is defined in this file so you don't need to rely on `Vaadin.registrations`.

File: test/details.test.js [hunk 2]
```diff
@@ -121,6 +127,10 @@ describe('vaadin-details', () => {
       expect(toggle.getAttribute('aria-expanded')).to.equal('true');
     });
 
+    it('should set aria-controls on toggle button', () => {
+      expect(toggle.getAttribute('aria-controls')).to.equal(`vaadin-details-content-${getUniqueId()}`);
```
Comment (web-padawan) [inline test/details.test.js:131]: Please consider using a RegEx here, same as in [vaadin-radio-group](https://github.com/vaadin/vaadin-radio-button/blob/ac80d5e6cac1b266c44da81bac79a1c13e930d15/test/radio-group.test.js#L541) tests.

File: test/details.test.js [hunk 3]
```diff
@@ -129,6 +139,25 @@ describe('vaadin-details', () => {
...
+          </vaadin-details>
+          <vaadin-details>
+            <div slot="summary">Summary</div>
+            <input>
+          </vaadin-details>
+        </div>
+      `);
+      const details = container.querySelectorAll('vaadin-details');
+      expect(details[0]._collapsible.id).to.equal(`vaadin-details-content-${id}`);
+      expect(details[1]._collapsible.id).to.equal(`vaadin-details-content-${1 + id}`);
```
Comment (web-padawan) [inline test/details.test.js:159]: No need to check for specific number here. 
It is enough to compare the `id` of two elements and check they are different.

**Review by web-padawan — CHANGES_REQUESTED** [review]

**Comment by niiyeboah** [comment]
Creating new PR for [web-components](https://github.com/vaadin/vaadin-web-components)

**Pull request closed without merging**